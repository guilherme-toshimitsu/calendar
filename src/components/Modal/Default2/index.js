export { default } from './Default2';
