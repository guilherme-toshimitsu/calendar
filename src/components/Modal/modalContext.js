import { createContext } from 'react';

const ModalContext = createContext({
  openModal: () => {},
  closeModal: () => {},
  pickModal: () => {},
});
const ModalProvider = ModalContext.Provider;
const ModalConsumer = ModalContext.Consumer;

export default ModalContext;
export { ModalConsumer, ModalProvider, ModalContext };
