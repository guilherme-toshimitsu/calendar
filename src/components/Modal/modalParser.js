import Default from './Default';
import Default2 from './Default2';

const parser = {
  default: Default,
  default2: Default2,
};

export default parser;
