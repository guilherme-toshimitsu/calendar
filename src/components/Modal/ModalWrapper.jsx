import React, {useReducer} from 'react';
import PropTypes from 'prop-types';

import {createReducer} from '@utils/createReducer';

import {ModalProvider} from './modalContext';
import Modal from './Modal';
import * as actionTypes from './Reducer/modalActionTypes';
import actions from './Reducer/modalActions';
import initialState from './Reducer/modalState';

const reducer = createReducer(actions);

const ModalWrapper = ({children}) => {
	const [state, dispatch] = useReducer(reducer, initialState);

	const actionResolver = {
		openModal: () => dispatch({type: actionTypes.OPEN_MODAL}),
		closeModal: () => dispatch({type: actionTypes.CLOSE_MODAL}),
		loadProps: (props) => dispatch({type: actionTypes.LOAD_PROPS, props}),
		selectModal: (selectedModal) =>
			dispatch({type: actionTypes.SELECT_MODAL, selectedModal}),
	};

	return (
		<ModalProvider value={[state, actionResolver]}>
			<Modal />
			{children}
		</ModalProvider>
	);
};

ModalWrapper.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.element,
		PropTypes.arrayOf(PropTypes.element),
	]).isRequired,
};

export default ModalWrapper;
