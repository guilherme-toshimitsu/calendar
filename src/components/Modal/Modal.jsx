import React from 'react';
import PropTypes from 'prop-types';
import { Dialog } from '@material-ui/core';
import { useModal } from '@components/Modal';
import modalParser from './modalParser.js';

const Modal = () => {
  const [{ selectedModal, props, open }, { closeModal }] = useModal();

  const CustomComponent = modalParser[selectedModal] ? modalParser[selectedModal] : modalParser['default'];
  return (
    <Dialog disableBackdropClick open={open} onClose={() => closeModal()}>
      <CustomComponent {...props} onClose={() => closeModal()} />
    </Dialog>
  );
};

Modal.defaultProps = {
  open: false,
};

Modal.propTypes = {
  open: PropTypes.bool,
};

export default Modal;
