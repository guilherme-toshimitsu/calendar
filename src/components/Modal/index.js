import { useContext } from 'react';
import Modal from './Modal';
import ModalWrapper from './ModalWrapper';
import { ModalConsumer, ModalProvider, ModalContext } from './modalContext';

const useModal = () => useContext(ModalContext);

export default Modal;
export { useModal, ModalWrapper, ModalConsumer, ModalProvider, ModalContext };
