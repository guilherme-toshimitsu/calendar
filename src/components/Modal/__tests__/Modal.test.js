import React from 'react';

import { render, fireEvent, wait } from '@testing-library/react';

import Modal, { ModalWrapper, useModal } from '../';

describe('<Modal> Modal Component', () => {
  it('render full modal', () => {
    const Teste = () => {
      const { openModal } = useModal();
      const clickOpen = () => openModal();
      return (
        <div>
          <button data-testid="button-test-modal" onClick={clickOpen}>
            "aqui"
          </button>
        </div>
      );
    };

    const component = render(
      <ModalWrapper>
        <Teste />
      </ModalWrapper>
    );

    expect(component.asFragment()).toMatchSnapshot();
  });
});
