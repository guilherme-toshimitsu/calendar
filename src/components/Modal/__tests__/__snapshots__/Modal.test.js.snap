// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal> Modal Component render full modal 1`] = `
<DocumentFragment>
  <div>
    <button
      data-testid="button-test-modal"
    >
      "aqui"
    </button>
  </div>
</DocumentFragment>
`;
